packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```
@@ -146,39 +72,287 @@ const TODO = (method: keyof Client) => {
+                      ]
+                    : [])
+            ]
+        }
+    },
     addGiftCertificateItemToBasket: TODO('addGiftCertificateItemToBasket'),
-    addItemToBasket: updateBasket,
-    addPaymentInstrumentToBasket: updateBasket,
+    addItemToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addPaymentInstrumentToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
     addPriceBooksToBasket: TODO('addPriceBooksToBasket'),
     addTaxesForBasket: TODO('addTaxesForBasket'),
     addTaxesForBasketItem: TODO('addTaxesForBasketItem'),
-    createBasket: updateBasketWithResponseBasketId,
+    createBasket(customerId, {parameters}, response) {
+        const {basketId} = response
+
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId && basketId
```

Should the logic here be a bit more complex? Update if `customerId && basketId`, but fall back to invalidate if just `customerId`.